quantmsrescore/ms2_model_manager.py

```
@@ -33,20 +73,75 @@ def __init__(self,
+        if os.path.exists(model_zip_file_path):
+            if not skip_if_exists:
+                raise FileExistsError(f"Model file already exists: {model_zip_file_path}")
+            logging.debug(f"Model file already exists, skipping download: {model_zip_file_path}")
+        else:
+            logging.info(f"Downloading pretrained models from {url} to {model_zip_file_path} ...")
+            try:
+                os.makedirs(os.path.dirname(model_zip_file_path), exist_ok=True)
+                context = ssl.create_default_context(cafile=certifi.where())
+                # Use streaming download with longer timeout for large model files
+                # timeout=300s (5 min) for slow connections; stream in 1MB chunks
+                with urllib.request.urlopen(url, context=context, timeout=300) as response:  # nosec B310
```

Ruff (0.14.10)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)